File: pushlib/src/main/java/com/saint/pushlib/jpush/JPushInit.java
package com.saint.pushlib.jpush;

import android.app.Activity;
import android.app.Application;

import com.saint.pushlib.BasePushInit;
import com.saint.pushlib.PushControl;
import com.saint.pushlib.R;
import com.saint.pushlib.PushConstant;
import com.saint.pushlib.bean.ReceiverInfo;
import com.saint.pushlib.receiver.PushReceiverManager;

import cn.jpush.android.api.JPushInterface;

public class JPushInit extends BasePushInit {

    /**
     * 推送初始化
     *
     * @param isDebug     设置debug模式
     * @param application --
     */
    public JPushInit(boolean isDebug, Application application) {
        super(isDebug, application);
        JPushInterface.setDebugMode(isDebug);
        JPushInterface.init(application);
        ReceiverInfo info = new ReceiverInfo();
        info.setPushType(PushConstant.JPUSH);
        info.setContent("极光初始化");
        info.setTitle(mContext.getString(R.string.init_succeed));
        PushReceiverManager.getInstance().onRegistration(application, info);
    }

    @Override
    public void loginIn(Activity activity) {
        super.loginIn(activity);
        ReceiverInfo info = new ReceiverInfo();
        info.setTitle(mContext.getString(R.string.get_token));
        info.setContent(JPushInterface.getRegistrationID(activity));
        info.setPushType(PushConstant.JPUSH);
        PushReceiverManager.getInstance().setToken(PushControl.getInstance().getApplication(), info);
    }

    @Override
    public void setAlias(String alias) {
        super.setAlias(alias);
        JPushInterface.setAlias(mContext, 0, alias);
    }

    @Override
    public void loginOut() {
        JPushInterface.deleteAlias(mContext, 0);
        ReceiverInfo aliasInfo = new ReceiverInfo();
        aliasInfo.setTitle(mContext.getString(R.string.tip_off_push_succeed));
        aliasInfo.setContent(mAlias);
        aliasInfo.setPushType(PushConstant.JPUSH);
        PushReceiverManager.getInstance().onLoginOut(mContext, aliasInfo);
    }
}

File: pushlib/src/main/java/com/saint/pushlib/jpush/JPushInit.kt
package com.saint.pushlib.jpush

import android.app.Application
import cn.jiguang.api.utils.JCollectionAuth
import cn.jpush.android.api.JPushInterface
import cn.jpush.android.data.JPushConfig
import com.saint.pushlib.BasePushInit
import com.saint.pushlib.PushConfig
import com.saint.pushlib.PushConstant
import com.saint.pushlib.PushConstant.JPUSH
import com.saint.pushlib.R
import com.saint.pushlib.bean.ReceiverInfo
import com.saint.pushlib.receiver.PushReceiverManager

class JPushInit(isDebug: Boolean, application: Application,pushConfig: PushConfig) : BasePushInit(isDebug, application) {
    override fun loginIn() {
        onToken(JPushInterface.getRegistrationID(mContext), JPUSH)
    }

    override fun setAlias(alias: String?) {
        super.setAlias(alias)
        JPushInterface.setAlias(mContext, 0, alias)
    }

    override fun loginOut() {
        JPushInterface.deleteAlias(mContext, 0)
        val aliasInfo = ReceiverInfo()
        aliasInfo.title = mContext.getString(R.string.tip_off_push_succeed)
        if (mAlias != null) aliasInfo.content = mAlias!!
        aliasInfo.pushType = PushConstant.JPUSH
        PushReceiverManager.onLoginOut(mContext, aliasInfo)
    }

    /**
     * 推送初始化
     *
     * @param isDebug     设置debug模式
     * @param application --
     */
    init {
        JPushInterface.setDebugMode(isDebug)
        JPushInterface.setSmartPushEnable(application,pushConfig.enableJPushSmart)
//        JPushInterface.setLbsEnable(application,pushConfig.enableJPushLBS)
        JCollectionAuth.enableAutoWakeup(application,pushConfig.enableJPushAutoWakeup)
        JPushInterface.init(application)
        JCollectionAuth.setAuth(mContext,true)
        initSucceed(getString(R.string.JPUSH), JPUSH)
    }

    override fun clearNotificationAll() {
        super.clearNotificationAll()
        JPushInterface.clearAllNotifications(mContext)
    }
}